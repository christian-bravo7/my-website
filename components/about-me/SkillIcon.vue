<template>
  <div class="flex flex-col group items-center justify-center w-full p-2 skill-icon hover:shadow-lg transition-shadow duration-150">
    <span class="w-16 text-pink-800 dark:text-blue-500">
      <slot name="default" />
    </span>
    <span class="mt-2 text-gray-300 dark:text-blue-700 group-hover:text-pink-800 dark-group-hover:text-blue-500 text-base md:text-lg font-bold transition-colors duration-150">
      {{ name }}
    </span>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator';

@Component
export default class SkillIcon extends Vue {
  @Prop({ type: String, required: true })
  readonly name!: string;
}
</script>

<style lang="scss" scoped>
.skill-icon {
  svg {
    @apply fill-current;
  }
}
</style>
