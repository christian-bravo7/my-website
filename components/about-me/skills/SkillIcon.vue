<template>
  <div class="flex flex-col items-center justify-between | p-2 | bg-gray-100 dark:bg-blue-700 | rounded-xl shadow-md hover:shadow-xl | select-none | group transition-shadow duration-300 | skill-icon">
    <span class="flex justify-center items-center flex-1 | w-16">
      <slot name="default" />
    </span>
    <span class="mt-2 | text-black dark:text-white group-hover:text-pink-500 dark-group-hover:text-blue-500 | text-base md:text-lg | font-bold | transition-colors duration-300">
      {{ name }}
    </span>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator';

@Component
export default class SkillIcon extends Vue {
  @Prop({ type: String, required: true })
  readonly name!: string;
}
</script>

<style lang="scss" scoped>
.skill-icon {
  min-width: 140px;

  svg {
    @apply fill-current;
  }
}
</style>
