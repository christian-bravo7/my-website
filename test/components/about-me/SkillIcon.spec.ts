import { mount } from '@vue/test-utils';

import SkillIcon from '@/components/about-me/SkillIcon.vue';

describe('SkillIcon component', () => {
  it('should render correctly', () => {
    const wrapper = mount(SkillIcon, {
      propsData: {
        name: 'This is an icon text',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should display name text as prop', () => {
    const wrapper = mount(SkillIcon, {
      propsData: {
        name: 'This is an icon text',
      },
    });

    const name = wrapper.find('span.text-base.font-bold');

    expect(name.text()).toBe('This is an icon text');
  });

  it('should render child component as default slot', () => {
    const wrapper = mount(SkillIcon, {
      propsData: {
        name: 'This is an icon text',
      },
      slots: {
        default: {
          template: '<h1>This is a default slot</h1>',
        },
      },
    });

    const slotContent = wrapper.find('h1');

    expect(slotContent.text()).toBe('This is a default slot');
  });
});
