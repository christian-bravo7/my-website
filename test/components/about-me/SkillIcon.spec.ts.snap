// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillIcon component should render correctly 1`] = `
<div class="flex flex-col group items-center justify-center w-full p-2 skill-icon hover:shadow-lg transition-shadow duration-150"><span class="w-16 text-pink-800 dark:text-blue-500"></span> <span class="mt-2 text-gray-300 dark:text-blue-700 group-hover:text-pink-800 dark-group-hover:text-blue-500 text-base md:text-lg font-bold transition-colors duration-150">
    This is an icon text
  </span></div>
`;
